Kudu.Core/Settings/DiagnosticsSettingsManager.cs

```diff
@@ -87,6 +93,7 @@ private void SaveSettings(DiagnosticsSettings diagnosticsSettings)
                 FileSystemHelpers.EnsureDirectory(Path.GetDirectoryName(_path));
             }
 
+            diagnosticsSettings.EnsureValid();
```

Comment:
RE: Won't the TraceEnum converter validate these settings for us on deserialization?

[shrimpy] See below "9999" value is also get pass deserialization, which we do not want it to happen. We will need to do a IsDefined check just for safety.

![image](https://cloud.githubusercontent.com/assets/79823/11005287/cba22792-8471-11e5-9a2e-d8ee8461d1e7.png)


RE: If not, we could have our own converter that does. That way, when we add new settings we don't have to remember to add the setting key name to your static list.


[shrimpy] do you mean to have a converter like "StringEnumConverter", and attribute on Property level?

the issue is when one setting value is wrong, we will also want to update its co-related setting values.

e.g if AzureTableTraceLevel is corrupted, we will want to reset AzureTableEnabled as well.
for a customer converter, we will need to enforce the order of deescalation, might a bit over engineering for this issue.